sw/device/lib/dif/dif_hmac_unittest.cc

```diff
@@ -0,0 +1,117 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+#include <array>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/testing/mock_mmio.h"
+#include "sw/device/lib/dif/dif_test_base.h"
+
+#include "hmac_regs.h"  // Generated
+
+namespace dif_hmac_unittest {
+
+using mock_mmio::MmioTest;
+using mock_mmio::MockDevice;
+using testing::Test;
+
+// Base class for the rest fixtures in this file.
+class HmacTest : public testing::Test, public mock_mmio::MmioTest {
+ protected:
+  dif_hmac_t hmac_;
+  dif_hmac_transaction_t transaction_ = {
+      .message_endianness = kDifHmacEndiannessLittle,
+      .digest_endianness = kDifHmacEndiannessLittle,
+  };
+
+  struct ConfigRegister {
+    bool hmac_enable = false;
+    bool sha_enable = true;
+    bool msg_little_endian = true;
+    bool digest_big_endian = false;
+  } config_reg_;
+
+  HmacTest() { EXPECT_DIF_OK(dif_hmac_init(dev().region(), &hmac_)); }
+
+  void ExpectConfig(void) {
+    EXPECT_WRITE32(
+        HMAC_CFG_REG_OFFSET,
+        {
+            {HMAC_CFG_HMAC_EN_BIT, config_reg_.hmac_enable},
+            {HMAC_CFG_SHA_EN_BIT, config_reg_.sha_enable},
+            {HMAC_CFG_ENDIAN_SWAP_BIT, config_reg_.msg_little_endian},
+            {HMAC_CFG_DIGEST_SWAP_BIT, config_reg_.digest_big_endian},
+        });
+  }
+
+  void ExpectKey(const uint8_t *key, size_t size) {
+    for (size_t i = 0; i < size; i += sizeof(uint32_t)) {
+      uint32_t word = 0;
+      memcpy(&word, &key[i], sizeof(uint32_t));
+      EXPECT_WRITE32(i + HMAC_KEY_0_REG_OFFSET, word);
+    }
+  }
+};
+
+class HmacMacTest : public HmacTest {
+ protected:
+  static constexpr std::array<uint8_t, 32> kKey_ = {
```

Comment:
`private_member_` and `kConst` don't mix. Use one or the other (`kKey` wins here).

Reply:
done